### ISSUE TITLE
Proxy/Router deadlock or stale cache

### ISSUE THREAD

**Issue by asim** [micro/development#110]
There is a situation in which the proxy stops functioning either because of some sort of router deadlock or because of some stale registry issues. Identify and fix this garbage.

**Comment by domwong** [comment]
@asim do you have any repro or logs etc for this?

**Comment by asim** [comment]
I do not have any logs for this. I think I tracked down and fixed the issue but there's maybe a more forward looking fix also.

So the issue as I see it. When the `micro network` was being restarted on the platform, the new copy was not yet running, so services when attempting to make a call to re-register with the `micro registry`  would fail. They had a register TTL of 60 seconds and register interval of 30 seconds, resulting in a sort of race condition which led to them being de-registered around the next time they went to re-register and effectively being removed from registry. 

I have increased the register TTL to 90 seconds and added retries where there is an existing failure to register. I think further fixes might be to add readiness checks to all core services so that we don't shutdown the old version until the new one is running and maybe to perform more optimal route caching inside the go-micro/router so that we don't immediately throwaway routes where we do see the events for deleting them from the registry.

I think it highlights the need for more robustness in the core networking. If routes vanish, we lose ability to call services. We may want to cascade out old entries over time rather than immediately.

Any thoughts welcome.
